src/Aspire.Hosting/Dcp/DcpHost.cs

```diff
@@ -278,6 +279,11 @@ private async Task LogSocketOutputAsync(Socket socket, CancellationToken cancell
             tab = line.IndexOf((byte)'\t');
             var category = line[..tab];
             line = line[(tab + 1)..];
+
+            // Trim trailing carriage return.
+            Debug.Assert(line[^1] == '\r', "Expected line to end with a carriage return.");
```

Comment:
Would it be better/more robust to turn this `assert` into an `if` check?

Reply:
I switched it to an if test. It's basically no overhead so it doesn't matter, and the code is a little more resilent to change.